From baa8aeda5735eb048df81a88e80a870c359f31ad Mon Sep 17 00:00:00 2001
From: Neil Armstrong <narmstrong@baylibre.com>
Date: Thu, 29 Aug 2019 17:23:42 +0200
Subject: [PATCH] FROMGIT: arm64: dts: meson-sm1-sei610: add
 keep-power-in-suspend property in SDIO node

The WiFi firmware requires that the power is kept enabled while in
suspend mode. Add the keep-power-in-suspend property in the SDIO node
to specify that the power must be kept when entering in a system wide
suspend state.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>
Signed-off-by: Kevin Hilman <khilman@baylibre.com>
(cherry picked from commit 2e09574d172e6da4ee251203660c5a56b0b4c068
 https://git.kernel.org/pub/scm/linux/kernel/git/khilman/linux-amlogic.git v5.5/dt64)
Change-Id: Idd515178a0e54f5d9f64928129dfde5a34f3d018
---
 arch/arm64/boot/dts/amlogic/meson-sm1-sei610.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-sei610.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-sei610.dts
index e6913869001e..aae9782690e9 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-sei610.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-sei610.dts
@@ -462,6 +462,9 @@
 	non-removable;
 	disable-wp;
 
+	/* WiFi firmware requires power to be kept while in suspend */
+	keep-power-in-suspend;
+
 	mmc-pwrseq = <&sdio_pwrseq>;
 
 	vmmc-supply = <&vddao_3v3>;
